arviz/plots/ecdfplot.py

```diff
@@ -0,0 +1,341 @@
+    For eg: Plot ecdf-difference plot with confidence bands for PIT of sample for
+    comparing a given sample w.r.t a given sample
+
+    .. plot::
+        :context: close-figs
+
+        >>> sample2 = norm(0,1).rvs(5000)
+        >>> az.plot_ecdf(sample, sample2, confidence_bands = True, difference = True, pit = True)
+
+    """
+    if values2 is None and cdf is None and confidence_bands is True:
+        raise ValueError("For confidence bands you need to specify values2 or the cdf")
+
+    if cdf is not None and values2 is not None:
+        raise ValueError("To compare sample you need either cdf or values2 and not both")
+
+    if values2 is None and cdf is None and pit is True:
+        raise ValueError("For PIT specify either cdf or values2")
+
+    if values2 is None and cdf is None and difference is True:
+        raise ValueError("For ECDF difference plot need either cdf or values2")
+
+    if values2 is not None:
+        values2 = np.ravel(values2)
+        values2.sort()
+
+    values = np.ravel(values)
+    values.sort()
+
+    n = len(values)  # number of samples
+    ## This block computes gamma and uses it to get the upper and lower confidence bands
+    ## Here we check if we want confidence bands or not
+    if confidence_bands:
+        ## If plotting PIT then we find the PIT values of sample.
+        ## Basically here we generate the evaluation points(x) and find the PIT values.
+        ## z is the evaluation point for our uniform distribution in compute_gamma()
+        if pit:
+            x = np.linspace(1 / npoints, 1, npoints)
+            z = x
+            ## Finding PIT for our sample
+            probs = cdf(values) if cdf else compute_ecdf(values2, values) / len(values2)
+        else:
+            ## If not PIT use sample for plots and for evaluation points(x) use equally spaced
+            ## points between minimum and maximum of sample
+            ## For z we have used cdf(x)
+            x = np.linspace(values[0], values[-1], npoints)
+            z = cdf(x) if cdf else compute_ecdf(values2, x)
+            probs = values
+
+        ## Computing gamma
+        if not pointwise:
+            gamma = compute_gamma(n, z, npoints, num_trials, fpr)
+        else:
+            gamma = fpr
+        ## Using gamma to get the confidence intervals
+        lower, higher = get_lims(gamma, n, z)
+
+        ## This block is for whether to plot ECDF or ECDF-difference
+        if not difference:
+            ## We store the coordinates of our ecdf in x_coord, y_coord
+            x_coord, y_coord = get_ecdf_points(x, probs, difference)
+        else:
+            ## Here we subtract the ecdf value as here we are plotting the ECDF-difference
+            x_coord, y_coord = get_ecdf_points(x, probs, difference)
+            for i, x_i in enumerate(x):
+                y_coord[i] = y_coord[i] - (
+                    x_i if pit else cdf(x_i) if cdf else compute_ecdf(values2, x_i)
+                )
```

Comment:
Loops like this will not be as fast as just computing the reference array and then subtracting it, because NumPy's vectorized functions will be much faster than Python for loops.

Reply:
This is something we can make faster in future PR if needed